bindings/java/src/main/java/org/ucx/jucx/ucp/UcpWorkerParams.java

```diff
@@ -0,0 +1,157 @@
+    public UcpWorkerParams requestThreadSafety() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+        this.threadMode = UcsConstants.ThreadMode.UCS_THREAD_MODE_MULTI;
+        return this;
+    }
+
+    /**
+     * Mask of which CPUs worker resources should preferably be allocated on.
+     * This value is optional.
+     * If it's not set, resources are allocated according to system's default policy.
+     */
+    public UcpWorkerParams setCpu(int cpuNum) {
+        this.fieldMask |=  UcpConstants.UCP_WORKER_PARAM_FIELD_CPU_MASK;
+        this.cpuMask.set(cpuNum);
+        return this;
+    }
+
+    /**
+     * Remote memory access send completion.
+     */
+    public UcpWorkerParams requestWakeupRMA() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_RMA;
+        return this;
+    }
+
+    /**
+     * Atomic operation send completion.
+     */
+    public UcpWorkerParams requestWakeupAMO() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_AMO;
+        return this;
+    }
+
+    /**
+     * Tag send completion.
+     */
+    public UcpWorkerParams requestWakeupTagSend() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_TAG_SEND;
+        return this;
+    }
+
+    /**
+     * Tag receive completion.
+     */
+    public UcpWorkerParams requestWakeupTagRecv() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_TAG_RECV;
+        return this;
+    }
+
+    /**
+     * This event type will generate an event on completion of any
+     * outgoing operation (complete or  partial, according to the
+     * underlying protocol) for any type of transfer (send, atomic, or RMA).
+     */
+    public UcpWorkerParams requestWakeupTX() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_TX;
+        return this;
+    }
+
+    /**
+     * This event type will generate an event on completion of any receive
+     * operation (complete or partial, according to the underlying protocol).
+     */
+    public UcpWorkerParams requestWakeupRX() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_RX;
+        return this;
+    }
+
+    /**
+     * Use edge-triggered wakeup. The event file descriptor will be signaled only
+     * for new events, rather than existing ones.
+     */
+    public UcpWorkerParams requestWakeupEdge() {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_EVENTS;
+        this.events |= UcpConstants.UCP_WAKEUP_EDGE;
+        return this;
+    }
+
+    /**
+     * User data associated with the current worker.
+     */
+    public UcpWorkerParams setUserData(ByteBuffer userData) {
```

Comment:
In java you can't get a pointer to an object. But you can allocate DirectByteBuffer (outside of the java heap) and in jni got a pointer to it: https://github.com/openucx/ucx/pull/3256/files#diff-ebf86b13a20f7edfd0cfa16f1243974cR55

Byte buffer provides different operations on it, e.g.: https://github.com/openucx/ucx/pull/3256/files#diff-cb0fa86fe6141c99d49847756c138b76R63

Reply:
not sure how it works with Java, but maybe we can make it a generic Object? so that user could attach any instance to it
e.g
```
   MyClass o = new MyClass();
  params.setUserData(o)
```

Reply:
` MyClass o = new MyClass();` is a heap object in java. You can't get a native pointer to it since after GC phase it'll be in another part of memory. DirectByteBuffer is not controlled by GC, so the pointer to it is valid. You can serialize any class to DirectByteBuffer.

Reply:
So in order to keep a reference to some arbitrary object in the heap, need to create some "proxy" class with Object field, and serialize it to ByteBuffer?

Reply:
https://netty.io/wiki/user-guide-for-4.x.html#writing-a-time-server
```To send a new message, we need to allocate a new buffer which will contain the message. We are going to write a 32-bit integer, and therefore we need a ByteBuf whose capacity is at least 4 bytes. Get the current ByteBufAllocator via ChannelHandlerContext.alloc() and allocate a new buffer.```
ByteBuffer is de-facto standart way of representing serialized objects in java, especially in network libraries.

Reply:
http://tutorials.jenkov.com/java-nio/channels.html
`Channels always read to, or write from, a Buffer.`

Reply:
```
So in order to keep a reference to some arbitrary object in the heap, need to create some "proxy" class with Object field, and serialize it to ByteBuffer?
```
In java it's usually done next way:
1. Construct your object.
2. Serialize it to ByteBuffer.
3. Send over the wire/save to file/ other IO.
4. Read ByteBuffer.
5. Deserialize it to object. 

Steps 1 and 5 could be even on other machines, so no need to keep reference to objects:
https://github.com/Mellanox/SparkRDMA/blob/master/src/main/scala/org/apache/spark/shuffle/rdma/RdmaRpcMsg.scala#L156

Reply:
@petro-rudenko the goal of endpoint user_data is to have a "user defined context" when an error handling callback is called. it's not sent to remote side..

Reply:
Yes and ByteBuffer is the most flexible for this suite. It allows communicating between Java  <-> C, you can serialize an object there, any other data (string, numbers, etc.).

Reply:
ok